### PR TITLE
Clear GC info when consuming GT_PUTARG_SPLIT nodes

### LINKED ISSUE #65395: codegenarmarch.cpp:3201 Assertion failed '(gcInfo.gcRegGCrefSetCur & killMask) == 0'
I incorrectly closed these as dups of https://github.com/dotnet/runtime/issues/65311
https://github.com/dotnet/runtime/issues/65283
https://github.com/dotnet/runtime/issues/65284
https://github.com/dotnet/runtime/issues/65285
https://github.com/dotnet/runtime/issues/65286

but it looks like it's a different assert:
```
Assert failure(PID 3755 [0x00000eab], Thread: 3755 [0x0eab]): Assertion failed '(gcInfo.gcRegGCrefSetCur & killMask) == 0' in 'System.Collections.Concurrent.ConcurrentDictionary`2[GenericLookupKey,__Canon][ILCompiler.DependencyAnalysis.ReadyToRunSymbolNodeFactory+GenericLookupKey,System.__Canon]:GrowTable(Tables[GenericLookupKey,__Canon]):this' during 'Generate code' (IL size 515)

File: /__w/1/s/src/coreclr/jit/codegenarmarch.cpp Line: 3201
```

### PR DESCRIPTION
**Pull request by jakobbotsch**
Fix #65395

### CODE REVIEW

**Comment by jakobbotsch** [comment]
Lots of failures due to "The response contained an invalid status code 500 Internal Server Error"... will retry the jitstressregs job.

**Review by kunalspathak — APPROVED** [review]
LGTM

File: src/coreclr/jit/codegenlinear.cpp [hunk 1]
```diff
@@ -1807,7 +1807,6 @@ void CodeGen::genConsumePutStructArgStk(GenTreePutArgStk* putArgNode,
 #if FEATURE_ARG_SPLIT
 //------------------------------------------------------------------------
 // genConsumeArgRegSplit: Consume register(s) in Call node to set split struct argument.
-//                        Liveness update for the PutArgSplit node is not needed
```
Comment (kunalspathak) [inline src/coreclr/jit/codegenlinear.cpp:1810]: Can you update the method name in the comment?

**Comment by jakobbotsch** [comment]
cc @dotnet/jit-contrib for another opportunity for comments/suggestions before I merge this.

File: src/coreclr/jit/codegenlinear.cpp (hunk 1 above)
Comment (jakobbotsch) [reply src/coreclr/jit/codegenlinear.cpp:1810]: Sure, but let me do it as part of another change to avoid rerunning CI.

**Review by BruceForstall — APPROVED** [review]

**Pull request merged**